### PR TITLE
Automated release process

### LINKED ISSUE #200: Implement GH Actions release process
See cylc/cylc-flow#3937

### PR DESCRIPTION
**Pull request by MetRonnie**
These changes close #200 

Tested: https://github.com/MetRonnie/cylc-uiserver/pull/1

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Review by kinow — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -5,8 +5,13 @@ all changes see the [closed
...
+<!-- The topmost release date is automatically updated by GitHub Actions. When
+creating a new release entry be sure to copy & paste the span tag with the
+`actions:bind` attribute, which is used by a regex to find the text to be
+updated. Only the first match gets replaced, so it's fine to leave the old
+ones in. -->
 -------------------------------------------------------------------------------
-## __cylc-uiserver-0.4 (2020-??-??)__
+## __cylc-uiserver-0.4.0 (<span actions:bind='release-date'>2021-04-?</span>)__
 
 ### Enhancements
```
Comment (kinow) [inline CHANGES.md:16]: Not related to this change, but I noticed on GitHub UI, having two paragraphs, one after the other without any text, results in the second paragraph being displayed "raw", without formatting.

image

The other entries have a single paragraph like "This is the release 1.2.bla".

File: .github/workflows/2_auto_publish_release.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        commitish: ${{ env.MERGE_SHA }}
+        tag_name: ${{ env.VERSION }}
+        release_name: cylc-uiserver-${{ env.VERSION }}
+        prerelease: ${{ env.PRERELEASE }}
+        body: |
+          See [${{ env.CHANGELOG_FILE }}](https://github.com/${{ github.repository }}/blob/master/${{ env.CHANGELOG_FILE }}) for detail.
+        # TODO: Get topmost changelog section somehow and use that as the body?
```
Comment (kinow) [inline .github/workflows/2_auto_publish_release.yml:58]: Also doesn't need to be addressed here, but here's my take (adapted from [this SO](https://stackoverflow.com/a/40450360)).

```
awk -F '[- ]' -v ver=0.4 '/^## / { if (p) { exit }; if ($4 == ver) { p=1; next } } p' CHANGES.md
```

File: CHANGES.md (hunk 1 above)
Comment (MetRonnie) [reply CHANGES.md]: Actually it was due to a non-breaking space after `###`

**Comment by MetRonnie** [comment]
Tested: https://github.com/MetRonnie/cylc-uiserver/pull/1

**Review by hjoliver — APPROVED** [review]

**Pull request merged**